src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java

```diff
@@ -220,32 +415,75 @@ private boolean isPlayerAttacking()
         // proof to new weapons which don't need custom code and the weapon
         // stats are enough.
         Actor target = client.getLocalPlayer().getInteracting();
-        if (target != null)
+        if (target != null && (target instanceof NPC))
         {
-            boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0;
+            // Turns out fishing spots look like target dummies... so check it's not one
+            boolean isNotFishingSpot = getIsNotFishingSpot(target);
+            boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0 && isNotFishingSpot;
             boolean attackingNPC = target.getCombatLevel() > 0 || isTargetDummy;
-            boolean notWalking = animationId != -1;
+            // Not walking is either any player animation or the edge cases which don't trigger an animation, e.g Salamander.
+            boolean notWalking = animationId != -1 || getSalamanderAttack();
             // just having a target is not enough the player may be out of
             // range, we must wait for any animation which isn't
             // running/walking/etc
             return attackingNPC && notWalking;
         }
-        // fallback - might as well see if the hard-coded list has this animation
-        return AnimationData.fromId(animationId) != null;
+        AnimationData fromId = AnimationData.fromId(animationId);
+        if (fromId == AnimationData.RANGED_BLOWPIPE || fromId == AnimationData.RANGED_BLAZING_BLOWPIPE)
+        {
+            // These two animations are the only ones which exceed the duration of their attack cooldown (when
+            // on rapid), so in this case DO NOT fall back the animation as it is un-reliable.
+            return false;
+        }
+        // fall back to animations.
+        return fromId != null;
+    }
+
+    private boolean getIsNotFishingSpot(Actor target) {
+        final NPC npc = (NPC) target;
+        FishingSpot spot = FishingSpot.findSpot(npc.getId());
+        return spot == null;
     }
```

Comment:
Most recent commit f1869aa | Saw someone raise this on the runelite discord:

> https://imgur.com/rcZoS5G
> any idea why this has started happening with attack timer metronome? haven't changed anything afaik

This small function fixes it:


https://github.com/ngraves95/attacktimer/assets/17512637/aa166886-11e8-4650-9253-16cc58f1cbc9